src/cmd/ksh93/bltins/typeset.c

```diff
@@ -1070,18 +1070,18 @@ int b_builtin(int argc, char *argv[], Shbltin_t *context) {
         sfputr(stkp, name, 0);
         errmsg = 0;
         addr = 0;
-        if (dlete || liblist) {
-            for (n = (nlib ? nlib : dlete); --n >= 0;) {
+        if (delete || nlib) {
+            for (n = (nlib ? nlib : delete ? 1 : 0); --n >= 0;) {
```

Comment:
Before this patch, could this line have been called with `dlete=2` ?

Reply:
Yes, and that's a bug. Because if nlib is zero or one you shouldn't execute the body of this loop with `n == 2`.